src/java.base/share/classes/java/text/ChoiceFormat.java

```diff
@@ -114,28 +115,46 @@
  * }
  * }
  * </blockquote>
- * <p>
- * Specifying a pattern for ChoiceFormat objects is fairly straightforward.
- * For example:
+ * Would output the following:
+ * <blockquote>
+ * <pre>{@code
+ * There are no files on ADisk
+ * There is one file on ADisk
+ * There are 2 files on ADisk
+ * There are 3 files on ADisk
+ * }</pre>
+ * </blockquote>
+ *
+ * <h2><a id="patterns">Patterns</a></h2>
+ * A {@code ChoiceFormat} pattern has the following syntax:
+ * <blockquote><pre>
+ * <i>Pattern:</i>
+ *         SubPattern *("|" SubPattern)
+ * <i>SubPattern:</i>
+ *         Limit Relation Format
+ * <i>Limit:</i>
+ *         {@code String} that can be parsed as a {@code double} / "&infin;" ({@code U+221E}) / "-&infin;" (-{@code U+221E}).
```

Comment:
Could the first choice be more specific? Also, do `SubPattern`s need to be sorted with `Limit`s?

Reply:
> Could the first choice be more specific?

Updated with a more syntactical definition. Please let me know if you think this is _too_ specific.

> Also, do SubPatterns need to be sorted with Limits?

If I am interpreting the question right, you're asking if a subPattern **needs** a limit to be valid. If that's the question, the answer is yes, `ChoiceFormat` is designed to have an equal amount of limits and formats (whether from the arrays, or the String pattern). But please let me know if I misinterpreted the question.

subpattern "xyz" where both _limit_ and _relation_ are omitted
```
jshell> var a = new ChoiceFormat("0#abc|xyz")
a ==> java.text.ChoiceFormat@17863
jshell> a.getLimits()
$21 ==> double[1] { 0.0 }
jshell> a.getFormats()
$22 ==> String[1] { "abc" }
```

subpattern "#xyz" where _limit_ is omitted
```
jshell> var b = new ChoiceFormat("0#abc|#xyz")
|  Exception java.lang.IllegalArgumentException: Each interval must contain a number before a format
```

Reply:
```
Please let me know if you think this is too specific.
```
I think the updated one is better.
For the second part, I meant:
```
jshell> new ChoiceFormat("0#def|-1#abc|xyz")
|  Exception java.lang.IllegalArgumentException: Incorrect order of intervals, must be in ascending order
```
This should be excluded from the pattern syntax

Reply:
> This should be excluded from the pattern syntax

I presume you meant that the syntax should _include_ the required ascending order of limits. Updated to make this apparent, I also adjusted the syntax a little bit to be more accurate. The previous syntax could technically have an empty _Number_